src/AddIns/Uno.UI.Lottie.Skia/ts/Uno.UI.Lottie.ts

```diff
@@ -0,0 +1,255 @@
+		properties: LottieAnimationProperties;
+	}
+
+	export class Lottie {
+		private static _player: LottiePlayer;
+		private static _runningAnimations: { [id: number]: RunningLottieAnimation } = {};
+		private static _numberOfFrames: number;
+
+		public static setAnimationProperties(newProperties: LottieAnimationProperties, animationData?: AnimationData): string {
+			const elementId = newProperties.elementId;
+
+			Lottie.withPlayer(p => {
+				let currentAnimation = Lottie._runningAnimations[elementId];
+
+				if (!currentAnimation || Lottie.needNewPlayerAnimation(currentAnimation.properties, newProperties)) {
+					// Here we need a new player animation
+					// (some property changes required a new animation)
+					currentAnimation = Lottie.createAnimation(newProperties, animationData);
+				}
+
+				Lottie.updateProperties(currentAnimation, newProperties);
+			});
+
+			return "ok";
+		}
+
+		public static stop(elementId: number): string {
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
+				a.stop();
+				Lottie.raiseState(a);
+			});
+
+			return "ok";
+		}
+
+		public static play(elementId: number, fromProgress: number, toProgress: number, looped: boolean): string {
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
+				a.loop = looped;
+
+				const fromFrame = fromProgress * Lottie._numberOfFrames;
+				const toFrame = toProgress * Lottie._numberOfFrames;
+
+				a.playSegments([fromFrame, toFrame], false);
+				Lottie.raiseState(a);
+			});
+
+			return "ok";
+		}
+
+		public static kill(elementId: number): string {
+			Lottie.withPlayer(p => {
+				Lottie._runningAnimations[elementId].animation.destroy();
+				delete Lottie._runningAnimations[elementId];
+			});
+
+			return "ok";
+		}
+
+		public static pause(elementId: number): string {
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
+				a.pause();
+				Lottie.raiseState(a);
+			});
+
+			return "ok";
+		}
+
+		public static resume(elementId: number): string {
+			Lottie.withPlayer(p => {
+				const a = Lottie._runningAnimations[elementId].animation;
+				a.play();
+				Lottie.raiseState(a);
+			});
+
+			return "ok";
+		}
+
+		public static setProgress(elementId: number, progress: number): string {
+			Lottie.withPlayer(p => {
+				const animation = Lottie._runningAnimations[elementId].animation;
+				const frames = animation.getDuration(true);
+				const frame = frames * progress;
+				animation.goToAndStop(frame, true);
+				Lottie.raiseState(animation);
+
+			});
+
+			return "ok";
+		}
+
+		public static getAnimationState(elementId: number): string {
+			const animation = Lottie._runningAnimations[elementId].animation;
+
+			const state = Lottie.getStateString(animation);
+
+			return state;
+		}
+
+		private static needNewPlayerAnimation(current: LottieAnimationProperties, newProperties: LottieAnimationProperties): boolean {
```

Comment:
Codacy found an issue: [Exceeds maximum line length of 120](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5872302)